### PR TITLE
FS-1094 message api call decommissioned from menu call

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

File: app/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageController.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Singleton()
+class ScaMessageController @Inject()(cc: ControllerComponents, messageConnector: MessageConnector, authenticate: AuthAction)
+                                    (implicit ec: ExecutionContext) extends BackendController(cc) with Logging {
+
+  def getUnreadMessageCount: Action[AnyContent] = authenticate.async { implicit request =>
+    logger.info(s"[ScaMessageController][getUnreadMessageCount]")
+
+    messageConnector.getUnreadMessageCount.map {
+      case Some(unreadMessageCount) => Ok(Json.toJson(unreadMessageCount))
+      case _ => NotFound
```
Comment (yangwong9664) [inline app/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageController.scala:38]: is NotFound an appropriate http response if there's no unread messages? I would lean towards 204

File: test/uk/gov/hmrc/singlecustomeraccountwrapperdata/connectors/MessageConnectorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+              .withHeader("Content-Type", "application/json")
+              .withBody(messageResponse.toString())
+          )
+      )
+      messageConnector.getUnreadMessageCount.map { response =>
+        response mustBe None
+      }
+    }
+  }
+}
```
Comment (yangwong9664) [inline test/uk/gov/hmrc/singlecustomeraccountwrapperdata/connectors/MessageConnectorSpec.scala:101]: can you add a unit test for the exception scenario?

File: test/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      val result = controller.getUnreadMessageCount(fakeRequest)
+
+      whenReady(result) { _ =>
+        status(result) mustBe OK
+        contentAsJson(result) mustBe Json.parse(unreadMessageCount.toString)
+      }
+
+    }
+
+    "return Not Found When nino is not valid" in {
```
Comment (yangwong9664) [inline test/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageControllerSpec.scala:73]: nino is not valid, is that right?

**Review by yangwong9664 — CHANGES_REQUESTED** [review]

File: build.sbt [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
   .enablePlugins(play.sbt.PlayScala, SbtDistributablesPlugin)
   .settings(
-        majorVersion        := 0,
-        scalaVersion        := "2.13.8",
-        libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test,
-        // https://www.scala-lang.org/2021/01/12/configuring-and-suppressing-warnings.html
-        // suppress warnings in generated routes files
-        scalacOptions += "-Wconf:src=routes/.*:s",
-        PlayKeys.playDefaultPort := 8422,
-        ScoverageKeys.coverageMinimumStmtTotal := 90,
```
[the remark this answers is not in the document]
Comment (cyeniceri) [reply build.sbt:15]: .settings(CodeCoverageSettings.settings: _*)
is providing CodeCoverageSettings already.

`
  val settings: Seq[Setting[_]] = Seq(
    ScoverageKeys.coverageExcludedPackages := excludedPackages.mkString(";"),
    ScoverageKeys.coverageMinimumStmtTotal := 90,
    ScoverageKeys.coverageFailOnMinimum := true,
    ScoverageKeys.coverageHighlighting := true
  )
`

File: app/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageController.scala (hunk 1 above)
Comment (cyeniceri) [reply app/uk/gov/hmrc/singlecustomeraccountwrapperdata/controllers/ScaMessageController.scala:38]: makes sense. Will change

**Review by kritikarai6 — APPROVED** [review]

**Pull request merged**